test/rules/use_string_buffer.dart

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N use_string_buffer`
+
+class A {
+  String buffer;
+
+  void foo(int n) {
+    int aux = n;
+    while(aux --> 0) {
```

Comment:
The formatting is strange here. I keep trying to read `-->` as an operator rather than reading the whole expression as `aux-- > 0`.